cores/esp8266/spiffs_api.h

```diff
@@ -154,6 +154,9 @@ class SPIFFSImpl : public FSImpl
             return;
         }
         SPIFFS_unmount(&_fs);
+        delete[] _workBuf.release();
```

Comment:
Why not do _workBuf.reset(nullptr), which is preferred?

Reply:
My understanding is that reset(x) does not implicitly call delete[], only delete.  https://en.cppreference.com/w/cpp/memory/unique_ptr/reset .

Reply:
The type of _workBuf is the specialization for dynamic arrays `std::unique_ptr<uint8_t[]>`, and so the default deleter object is `delete[]`. This is actually why, if you're going to delete manually, it is preferred to use `_workBuf.get_deleter()(_workBuf.release())`, at least for normal usage of unique_ptr.

The case that is dangerous is when you misuse unique_ptr, like when you don't use the specialization, e.g.:
```
uint8_t *raw = new uint8_t[256];
std::unique_ptr<uint8_t> buf(&raw[0]); //gain ownership
buf.reset(); //calls delete instead of delete[] => mem leak
```
in which case you *do* want to do:
`delete[] buf.release();`

About reset():
> 4) In the specialization for dynamic arrays, std::unique_ptr<T[]>, this overload is necessary to allow reset to nullptr (which would otherwise be prohibited by the template overload). Equivalent to reset(pointer())

The specialization is provided for dynamic arrays for the case when you pass in nullptr as well. As of C++17, you can even omit nullptr, as it's defaulted if you call without args. 
Per my understanding, the reason that this is needed is that the dynamic array type is T[], but you're passing in nullptr, which is not arrayed, so a naive implementation of unique_ptr would not allow nullptr as arg (different type, notice the deleted reset(U) which is disallowed because the arg type is U and not U[]).

If you want to put the doubt gremlins to rest, try a small test with the above and run it through valgrind. There's always the chance that I'm wrong 😄